TUMFARSynchrony/experimental-hub
# Ed websocket lobby redirect

# Overview
TODO: What does this PR contain? Why was the change necessary?

Issue Linked: https://github.com/TUMFARSynchrony/experimental-hub/issues/134

## Changes
- Implemented a WebSocket server that broadcasts a message to all connected clients when an experimenter joins the lobby.
- Updated the client-side JavaScript to handle WebSocket messages and trigger a browser refresh upon receiving a specific message indicating a new experimenter's arrival.
- Refactored lobby management code to integrate WebSocket connection setup and message handling efficiently.

## How to test?
1. Start the server and navigate to the experimenter's lobby.
2. Open multiple browser windows as different participants and connect them to the lobby.
3. Join the lobby as an experimenter from another browser window.
4. Observe that all participant browsers automatically refresh upon the experimenter's entry.
5. Check the server logs to ensure that the WebSocket message is broadcasted correctly when the experimenter joins.

# Checklist
- [ X ] Correct base branch was selected (usually `development`)
- [ X ] `development` was merged into this branch and potential conflicts resolved
- [ X ] [Issues](https://github.com/TUMFARSynchrony/experimental-hub/issues) which were solved are linked. Be sure to make that manually because Closes/Fixes keywords only work when base branch is main.
- [ X ] Terminology is consistent with the rest of the codebase
- [ X ] Reviewers are assigned
- [ X ] Optional: Wiki pages were updated to reflect the changes (list changed pages above)
